Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h

```diff
@@ -2176,6 +2171,27 @@ class Surface_mesh
     }
     return is.good();
   }
+/// @endcond
+
+  /// \relates Surface_mesh
+  /// Extracts the surface mesh from an input stream in Ascii OFF, COFF, NOFF, CNOFF format.
+  /// The operator reads the point property as well as "v:normal", "v:color", and "f:color".
+  /// Vertex texture coordinates are ignored.
+  /// \pre `operator>>(std::istream&,const P&)` must be defined.
+
+  template <typename P>
+  bool read_off(std::istream& is, Surface_mesh<P>& sm)
+  {
+    if( (sm.num_vertices() == 0) && (sm.num_halfedges() == 0)&& (sm.num_faces() == 0) ){
+      return read_off_clear(is,sm);
+    }
+    Surface_mesh<P> other;
+    if(read_off_clear(is,other)){
+      copy_face_graph(other,sm);
```

Comment:
I think you can avoid this copy. In the read_off function, you can store in a std::vector the translated index of each index vertex read in the off file. You initialize this vector when you read the vertices; and use this translated table when you create each face.